### PR TITLE
commandExecutionResult event should be logged after afterCommandExecute hook

### LINKED ISSUE #7944: commandExecutionResult event should be logged after `afterCommandExecute` hook
### Expected behavior
- `commandExecutionResult` event should be the last event for transaction execution

### Actual behavior
Dry run shows
```
{
      "result":0,
      "events":[
         {
            "data":"0a14cdac35270a8bcb46552caffcbfa7dbcc1c87fcca12036665651a08040000000000000020f0900a2800",
            "index":0,
            "module":"token",
            "name":"lock",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683",
               "cdac35270a8bcb46552caffcbfa7dbcc1c87fcca"
            ],
            "height":379
         },
         {
            "data":"",
            "index":1,
            "module":"fee",
            "name":"insufficientFee",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683"
            ],
            "height":379
         },
         {
            "data":"0800",
            "index":2,
            "module":"token",
            "name":"commandExecutionResult",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683"
            ],
            "height":379
         },
         {
            "data":"0a14cdac35270a8bcb46552caffcbfa7dbcc1c87fcca12036665651a08040000000000000020f0900a2800",
            "index":3,
            "module":"token",
            "name":"unlock",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683",
               "cdac35270a8bcb46552caffcbfa7dbcc1c87fcca"
            ],
            "height":379
         },
         {
            "data":"0a14cdac35270a8bcb46552caffcbfa7dbcc1c87fcca1208040000000000000018f0900a2000",
            "index":4,
            "module":"token",
            "name":"burn",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683",
               "cdac35270a8bcb46552caffcbfa7dbcc1c87fcca"
            ],
            "height":379
         },
         {
            "data":"0a14cdac35270a8bcb46552caffcbfa7dbcc1c87fcca121483eb47425f89f534184b4e40663d72843f5da1d91a08040000000000000020002800",
            "index":5,
            "module":"token",
            "name":"transfer",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683",
               "cdac35270a8bcb46552caffcbfa7dbcc1c87fcca",
               "83eb47425f89f534184b4e40663d72843f5da1d9"
            ],
            "height":379
         },
         {
            "data":"0a14cdac35270a8bcb46552caffcbfa7dbcc1c87fcca121483eb47425f89f534184b4e40663d72843f5da1d918f0900a2000",
            "index":6,
            "module":"fee",
            "name":"generatorFeeProcessed",
            "topics":[
               "00f53939f904231e7875361f095c57263ddaa78de6d8e367817b916edc954683",
               "cdac35270a8bcb46552caffcbfa7dbcc1c87fcca",
               "83eb47425f89f534184b4e40663d72843f5da1d9"
            ],
            "height":379
         }
      ]
```

### Steps to reproduce
Call dry run on transaction

### Which version(s) does this affect? (Environment, OS, etc...)
6.0.0-alpha.9

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #7944 

### How was it solved?

Change order of events

### How was it tested?

Update unit test

### CODE REVIEW

File: framework/src/state_machine/state_machine.ts [hunk 1]
```diff
@@ -205,9 +193,21 @@ export class StateMachine {
 					this._logger.debug({ moduleName: mod.name }, 'Executing afterCommandExecute');
 					await mod.afterCommandExecute(transactionContext);
 					this._logger.debug({ moduleName: mod.name }, 'Executed afterCommandExecute');
+					ctx.eventQueue.unsafeAdd(
```
Comment (shuse2) [inline framework/src/state_machine/state_machine.ts:196]: standard event should be added based on the result of the execution.
if transaction is invalid, the event should not be added

**Comment by mosmartin** [comment]
Hi @Incede, this is not part of the PR (and you don't have to impl it 🙂 just FYI) but I noticed the import section of `framework/test/unit/state_machine/custom_modules.ts` could be cleaned up 
```typescript
import { BaseMethod, BaseCommand, BaseEndpoint, BaseModule } from '../../../src';
import { TransactionVerifyResult } from '../../../src/abi';
import { ModuleMetadata } from '../../../src/modules/base_module';
import {
	BlockAfterExecuteContext,
	BlockExecuteContext,
	GenesisBlockExecuteContext,
	TransactionExecuteContext,
} from '../../../src/state_machine';
```
to
```typescript
import {
	BaseMethod,
	BaseCommand,
	BaseEndpoint,
	BaseModule,
	TransactionVerifyResult,
	ModuleMetadata,
	BlockAfterExecuteContext,
	BlockExecuteContext,
	GenesisBlockExecuteContext,
	TransactionExecuteContext,
} from '../../../src';
```
since all the exports have been aggregated in the index file at `../../../src`

**Comment by Incede** [comment]
> Hi @Incede, this is not part of the PR (and you don't have to impl it 🙂 just FYI) but I noticed the import section of `framework/test/unit/state_machine/custom_modules.ts` could be cleaned up
> 
> ```ts
> import { BaseMethod, BaseCommand, BaseEndpoint, BaseModule } from '../../../src';
> import { TransactionVerifyResult } from '../../../src/abi';
> import { ModuleMetadata } from '../../../src/modules/base_module';
> import {
> 	BlockAfterExecuteContext,
> 	BlockExecuteContext,
> 	GenesisBlockExecuteContext,
> 	TransactionExecuteContext,
> } from '../../../src/state_machine';
> ```
> 
> to
> 
> ```ts
> import {
> 	BaseMethod,
> 	BaseCommand,
> 	BaseEndpoint,
> 	BaseModule,
> 	TransactionVerifyResult,
> 	ModuleMetadata,
> 	BlockAfterExecuteContext,
> 	BlockExecuteContext,
> 	GenesisBlockExecuteContext,
> 	TransactionExecuteContext,
> } from '../../../src';
> ```
> 
> since all the exports have been aggregated in the index file at `../../../src`
> 
> Also, since this is a unit test file, maybe you could update the filename to follow the conventon `*.spec.ts?` 🙂

Cleaned up the imports. Kept the file name unchanged as discussed.

File: framework/src/state_machine/state_machine.ts [hunk 2]
```diff
@@ -217,6 +205,22 @@ export class StateMachine {
...
+			);
+		} else {
+			ctx.eventQueue.unsafeAdd(
+				ctx.transaction.module,
+				EVENT_TRANSACTION_NAME,
+				codec.encode(standardEventDataSchema, { success: false }),
+				[ctx.transaction.id],
+			);
+		}
+
```
Comment (shuse2) [inline framework/src/state_machine/state_machine.ts:223]:
```suggestion
		ctx.eventQueue.unsafeAdd(
			ctx.transaction.module,
			EVENT_TRANSACTION_NAME,
			codec.encode(standardEventDataSchema, { success: status ===  TransactionExecutionResult.OK }),
			[ctx.transaction.id],
		);
```

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

**Review by mosmartin — APPROVED** [review]

**Pull request merged**